### DISCUSSION TITLE
Is there an API to get if all selectable rows are selected?

### DISCUSSION THREAD

**Discussion by danteissaias** [TanStack/table#4801] [Q&A]
I enable selection conditionally for certain rows. I want to know if all selectable rows are selected. This is my workaround, is there a way to do with with TanStack Table's API?

```typescript
const selectableRows = table
  .getRowModel()
  .flatRows.filter((row) => row.getCanSelect());
const { flatRows } = table.getSelectedRowModel();

return (
  <Checkbox
    checked={
      flatRows.length === selectableRows.length
        ? true
        : flatRows.length > 0 
        ? "indeterminate"
        : false
    }   
    onCheckedChange={() => table.toggleAllRowsSelected()}
  />  
);
  ```

**Comment by jonahallibone** [comment]
[`getIsAllRowsSelected()`](https://tanstack.com/table/v8/docs/api/features/row-selection#getisallrowsselected)

> ```ts
> getIsAllRowsSelected: () => boolean
> ``` 
> Returns whether or not all rows in the table are selected.


Is this what you're looking for?

```tsx
const selectableRows = table
  .getRowModel()
  .flatRows.filter((row) => row.getCanSelect());

const { flatRows } = table.getSelectedRowModel();

const getIsChecked = () => {
  if(table.getIsAllRowsSelected())
   return true
  }

  if(table.getIsSomeRowsSelected()) {
    return "indeterminate"
  }
 
  return false
}

return (
  <Checkbox
    checked={getIsChecked()}
    onCheckedChange={() => table.toggleAllRowsSelected()}
  />  
);
```

**Reply by danteissaias** [reply]
If I have 4 rows however 1 has selection disabled, that function will return false if all those 3 selectable rows are selected.

**Reply by jonahallibone** [reply]
Are you using `enableRowSelection`?
https://tanstack.com/table/v8/docs/api/features/row-selection#enablerowselection

**Reply by danteissaias** [reply]
Yes, but as a function.

**Reply by jonahallibone** [reply]
I'm not sure then. Could be a bug. Perhaps raise an issue?
